### ISSUE TITLE
rule: aria-allowed-children

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#925]
This one really surprised me, but there doesn't seem to be a rule that checks that an element is allowed certain children. For instance, the following isn't flagged by Axe:

```html
<div role="listbox">
  <p role="button">button</p>
  <p role="option">option</p>
</div>
```

**Comment by WilcoFiers** [comment]
Nope, that rule checks that elements are allowed a certain role. So for instance an `h1` isn't allowed to have `role=button`.

**Comment by straker** [comment]
See https://github.com/w3c/aria/issues/1033

**Comment by WilcoFiers** [comment]
This rule is a duplicate of the more recent https://github.com/dequelabs/axe-core/issues/2076
